widget/text_test.go

```diff
@@ -706,3 +706,106 @@ func TestText_lineBounds_variable_char_width(t *testing.T) {
 		})
 	}
 }
+
+func TestText_binarySearch(t *testing.T) {
+	tests := []struct {
+		name string
+		text string
+		want int
+	}{
+		{
+			name: "IM_Test",
+			text: "iiiiiiiiiimmmmmmmmmm",
+			want: 12,
+		},
+		{
+			name: "Single_Line",
+			text: "foobar foobar",
+			want: 9,
+		},
+		{
+			name: "WH_Test",
+			text: "wwwww hhhhhh",
+			want: 6,
+		},
+		{
+			name: "DS_Test",
+			text: "dddddd sssssss",
+			want: 8,
+		},
+		{
+			name: "DI_Test",
+			text: "dididi dididd",
+			want: 10,
+		},
+		{
+			name: "XW_Test",
+			text: "xwxwxwxw xwxw",
+			want: 7,
+		},
+		{
+			name: "W_Test",
+			text: "WWWWW",
+			want: 4,
+		},
+		{
+			name: "Empty",
+			text: "",
+			want: 0,
+		},
+	}
+	textSize := 10
+	textStyle := fyne.TextStyle{}
+	measurer := func(text []rune) int {
+		return fyne.MeasureText(string(text), textSize, textStyle).Width
+	}
+
+	for _, tt := range tests {
+		binarySearch := newBinarySearch(measurer, []rune(tt.text), 50)
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, tt.want, binarySearch(0, len(tt.text)))
+		})
+	}
+
+}
+
+func TestText_findSpaceIndex(t *testing.T) {
+	tests := []struct {
+		name string
+		text string
+		last int
+		want int
+	}{
+		{
+			name: "case_1",
```

Comment:
"case_n" is a poor naming scheme, what is it that you are testing?

case_1 is testing that findSpaceIndex returns the fallback if the string has no spaces, so how about "no_space_fallback"?
case_2 is testing that findSpaceIndex returns the correct index, so "single_space"
case_3 is testing that findSpaceIndex returns the last (second) index, so "double_space"
case_4 is testing that findSpaceIndex returns the last (third) index, so "last_space", "third_space", or "many_spaces"

Reply:
If you touch it anyways: using `map[string]struct{…}` for table tests instead of `[]struct{name string, …}` gives a better readability _and_ runs the tests in random order (which is good to detect order dependencies).

Like so:
```go
for name, tt := range map[string]struct {
	…
} {
	"test description": { … },
} {
	t.Run(name, func(t *testing.T) {
		…
	})
}
```